### PR TITLE
Attach and detach visualisations to nodes.

### PR DESCRIPTION
**Pull request by Josef-Vonasek**
### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md), [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: src/rust/ide/src/model/execution_context.rs [hunk 1]
```diff
@@ -39,6 +47,24 @@ pub struct LocalCall {
     pub definition : DefinitionId,
 }
 
+pub type VisualisationId = Uuid;
+
+#[derive(Clone,Debug)]
+pub struct Visualisation {
+    /// Node that is to be visualized.
+    pub node_id: Uuid,
```
Comment (farmaazon) [inline src/rust/ide/src/model/execution_context.rs:55]: for identifying nodes we usually use alias `ast::Id`

File: src/rust/ide/src/model/execution_context.rs [hunk 2]
```diff
@@ -58,16 +84,20 @@ pub type Id  = language_server::ContextId;
...
 pub struct ExecutionContext {
     /// A name of definition which is a root call of this context.
-    pub entry_point : DefinitionName,
-    stack           : RefCell<Vec<LocalCall>>,
+    pub entry_point: DefinitionName,
+    /// Local call stack.
+    stack: RefCell<Vec<LocalCall>>,
+    /// Set of active visualisations.
+    visualisations: RefCell<HashMap<VisualisationId,Visualisation>>,
     //TODO[ao] I think we can put here info about visualisation set as well.
```
Comment (farmaazon) [inline src/rust/ide/src/model/execution_context.rs:92]: I think this TODO can be removed now.

File: src/rust/ide/src/model/execution_context.rs [hunk 3]
```diff
@@ -17,6 +20,11 @@ use enso_protocol::language_server;
 #[fail(display="Tried to pop an entry point")]
 pub struct PopOnEmptyStack {}
 
+/// Error when using an Id that does not correspond to any known visualization.
+#[derive(Clone,Copy,Debug,Fail)]
+#[fail(display="Tried to use incorrect visualization Id")]
```
Comment (wdanilo) [inline src/rust/ide/src/model/execution_context.rs:25]: Errors should be sentences for users. Thus, they need dot on the end.

File: src/rust/ide/src/model/execution_context.rs [hunk 4]
```diff
@@ -82,6 +117,16 @@ impl ExecutionContext {
...
     }
 
+    /// Attaches a new visualization for current execution context.
+    pub fn attach_visualization(&self, vis: Visualization) {
+        self.visualizations.borrow_mut().insert(vis.id,vis);
+    }
+
+    /// Detaches visualization from current execution context.
+    pub fn detach_visualization(&self, id:&VisualizationId) -> FallibleResult<Visualization> {
+        Ok(self.visualizations.borrow_mut().remove(id).ok_or(InvalidVisualizationId{})?)
```
Comment (wdanilo) [inline src/rust/ide/src/model/execution_context.rs:127]: Replace `InvalidVisualizationId{}` with `InvalidVisualizationId`

File: src/rust/ide/src/model/synchronized/execution_context.rs [hunk 5]
```diff
@@ -219,4 +246,34 @@ mod test {
...
+
+    #[test]
+    fn attaching_visualizations() {
+        let exe_id   = model::execution_context::Id::new_v4();
+        let path     = controller::module::Path::from_module_name("Test");
+        let root_def = DefinitionName::new_plain("main");
+        let model    = model::ExecutionContext::new(root_def);
+        let ls       = language_server::MockClient::default();
+        let vis      = Visualization {
+            id: model::execution_context::VisualizationId::new_v4(),
```
Comment (wdanilo) [inline src/rust/ide/src/model/synchronized/execution_context.rs:258]: the name `new_v4` is enigmatic. What does it do? Is it our API? If its our, thisshould be changed. If its not, we should have wrapper around that. May be not part of this PR, but then add task for it pls.

Comment (Josef-Vonasek) [reply src/rust/ide/src/model/synchronized/execution_context.rs:258]: This pattern is used on all IDs, since they are just type aliases instead of newtypes.

File: src/rust/ide/src/model/execution_context.rs (hunk 3 above)
Comment (Josef-Vonasek) [reply src/rust/ide/src/model/execution_context.rs:25]: As far a I remember, after this message comes `:`, and more elaborate error message. So it's displayed as valid sentence.
